### DISCUSSION TITLE
Increase hit-testable size of GridSplitter without affecting layout?

### DISCUSSION THREAD

**Discussion by MechWarrior99** [AvaloniaUI/Avalonia#17694] [Q&A] [answered]
Hello, I have a GridSplitter that I would like to visually be thin (1 or 2 pixels), but I want the hit area to be larger (6 - 10 pixels). This is a pretty common behavior in most applications, but I am having trouble figuring out how to go about it.

Any suggestions would be most welcome, thanks!

**Comment by LaurentInSeattle** [comment]
This XAML below was "my" solution to this problem, but...  for WPF.
I think it could be ported to Avalonia, possibly using ControlThemes ? 

HTH 

    <UserControl.Resources>
        <Style x:Key="VerticalGridSplitter" TargetType="{x:Type GridSplitter}">
            <Setter Property="HorizontalAlignment" Value="Stretch"/>
            <Setter Property="VerticalAlignment" Value="Stretch"/>
            <Setter Property="Opacity" Value="0.7"/>
            <Setter Property="Background" Value="{StaticResource TabForeroundBrush}"/>
            <Setter Property="Panel.ZIndex" Value="999" />
            <Setter Property="Template">
                <Setter.Value>
                    <ControlTemplate TargetType="{x:Type GridSplitter}">
                        <Border Margin="-12,0" Background="Transparent">
                            <TextBlock x:Name="tb" Width="3"  Background="{TemplateBinding Background}" />
                        </Border>
                        <ControlTemplate.Triggers>
                            <Trigger Property="IsMouseOver" Value="True">
                                <Setter Property="Background" TargetName="tb" Value="DarkOrange" />
                            </Trigger>
                        </ControlTemplate.Triggers>
                    </ControlTemplate>
                </Setter.Value>
            </Setter>
        </Style>
        <Style x:Key="HorizontalGridSplitter" TargetType="{x:Type GridSplitter}">
            <Setter Property="HorizontalAlignment" Value="Stretch"/>
            <Setter Property="VerticalAlignment" Value="Stretch"/>
            <Setter Property="Opacity" Value="0.7"/>
            <Setter Property="Background" Value="{StaticResource TabForeroundBrush}"/>
            <Setter Property="Panel.ZIndex" Value="999" />
            <Setter Property="Template">
                <Setter.Value>
                    <ControlTemplate TargetType="{x:Type GridSplitter}">
                        <Border Margin="0,-12" Background="Transparent" IsHitTestVisible="True">
                            <TextBlock x:Name="tb" Width="9999" Height="3" Text="kjklj" Background="{TemplateBinding Background}" />
                        </Border>
                        <ControlTemplate.Triggers>
                            <Trigger Property="IsMouseOver" Value="True">
                                <Setter Property="Background" TargetName="tb" Value="DarkOrange" />
                            </Trigger>
                        </ControlTemplate.Triggers>
                    </ControlTemplate>
                </Setter.Value>
            </Setter>
        </Style>
    </UserControl.Resources>

**Reply by MechWarrior99** [reply]
That was kind of my thinking of how to go about it as well. Unfortunatly it doesn't seem to get hit. This does feel like the right way to go about it though. :/

**Reply by LaurentInSeattle** [reply]
No hit ? Note: Stuff with no background does not get pointer events in Avalonia...

**Reply by MechWarrior99** [reply]
No hit, yeah I spent too long one time trying to debug an issue before I realized there has to be a background, never forgot after that haha. So I made sure it had a background (trasparent), and HitTestVisible.

**Reply by MechWarrior99** [reply]
Thanks I figured it out, needed to disable ClipToBounds. Which in hinsight, makes sense, for some reason I thought it was disabled by default.

**Comment by MechWarrior99** [comment, marked as answer]
I figured out the answer, the idea is two have a `Border` nested inside of another `Border`, with the outer border being the selection area, and the smaller one being the actual visual area so to speak.
A couple of important factors, the `ZIndex` may need to be high to ensure it is above the grid elements. And most importantly, `ClipToBounds` but be false, otherwise the extra selection area will be clipped.

Here is a tiny 'hard-coded' example for a horizontal grid splitter.
```xml
 <Style Selector="GridSplitter.wide">
            <Setter Property="ClipToBounds" Value="False"/>
            <Setter Property="Template">
                <ControlTemplate>
                    <Border Width="20" Background="Transparent" IsHitTestVisible="True">
                        <Border BorderBrush="{TemplateBinding BorderBrush}"
                                BorderThickness="{TemplateBinding BorderThickness}"
                                CornerRadius="{TemplateBinding CornerRadius}"
                                Background="{TemplateBinding Background}"
                                HorizontalAlignment="Center"
                                Width="2"/>
                    </Border>
                </ControlTemplate>
            </Setter>
        </Style>
        ```
